py3status/modules/usbguard.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+"""
+Usbguard i3 companion to replace default 'usbguard-applet-qt'.
+Show an indicator when a new usb device is plugged,
+we can allow or reject a device.
+
+Configuration parameters:
+    allow_urgent: Allow urgent (default False)
+    button_allow: Button to allow the device. (default 1)
+    button_block: Button to block the device (default None)
+    button_reject: Button to reject the device. (default 3)
+    format: Display format for the module. (default '[UsbGuard: [{device_name}]]')
+
+Format placeholders:
+    {device_name} the device name of last device plugged.
+    {device_id} the usbguard device id of last device plugged.
+    {device_hash} the usbguard device unique hash of last device plugged.
+
+Requires:
+    https://usbguard.github.io:
+    "The USBGuard software framework helps to protect your
+    computer against rogue USB devices..."
+
+@author Cyril Levis (@cyrinux)
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'Usbguard: Mass Storage', 'urgent': True}
+"""
+
+import threading
+from time import sleep
+from gi.repository import GLib
+from pydbus import SystemBus
+
+DBUS_INTERFACE = 'org.usbguard'
+DBUS_DEVICE_FILTER = '/org/usbguard/Devices'
+
+
+class UsbguardListener(threading.Thread):
+    def __init__(self, parent):
+        super(UsbguardListener, self).__init__()
+        self.parent = parent
+
+    def _setup_dbus(self):
+        self.dbus = SystemBus()
+        self.dbus_devices_filter = DBUS_DEVICE_FILTER
+        try:
+            self.parent.error = None
+            self.proxy = self.dbus.get(DBUS_INTERFACE)
+            self.proxy_devices = self.proxy[".Devices"]
+        except:
+            self.parent.error = Exception("usbguard-dbus service not running")
+
+    # on device change signal
+    def _cb_devices_presence_changed(self, *args):
+        device = self.parent.device
+        device_args = args[4]
+        device_hash = device_args[4]['hash']
+        device_id = device_args[0]
+        device_name = device_args[4]['name']
+        device_perms = device_args[3]
+        device_status = device_args[1]
+        device['device_name'] = None
```

Comment:
(Improv) This had to do something with `device` (obviously). We could make this readability to remove repetitive `device` as it is slightly hard to track parameters. P.S. What does `args` show and what does `_cb_` stand for?

Reply:
_cb was for callback, I have change this,
About args, I'll think of this ok :)

Reply:
pprint args inside 2 callback show this structure
```
(':1.7',
 '/org/usbguard/Devices',
 'org.usbguard.Devices',
 'DevicePresenceChanged',
 (43,
  3,
  0,
  'allow id 1050:0407 serial "" name "Yubikey 4 OTP+U2F+CCID" hash '
  '"hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=" parent-hash '
  '"jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=" via-port "1-3" '
  'with-interface { 03:01:01 03:00:00 0b:00:00 }',
  {'hash': 'hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=',
   'id': '1050:0407',
   'name': 'Yubikey 4 OTP+U2F+CCID',
   'parent-hash': 'jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=',
   'serial': '',
   'via-port': '1-3',
   'with-interface': '03:01:01 03:00:00 0b:00:00'}))

(':1.7',
 '/org/usbguard/Devices',
 'org.usbguard.Devices',
 'DevicePresenceChanged',
 (45,
  1,
  1,
  'block id 1050:0407 serial "" name "Yubikey 4 OTP+U2F+CCID" hash '
  '"hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=" parent-hash '
  '"jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=" via-port "1-3" '
  'with-interface { 03:01:01 03:00:00 0b:00:00 }',
  {'hash': 'hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=',
   'id': '1050:0407',
   'name': 'Yubikey 4 OTP+U2F+CCID',
   'parent-hash': 'jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=',
   'serial': '',
   'via-port': '1-3',
   'with-interface': '03:01:01 03:00:00 0b:00:00'}))

(':1.7',
 '/org/usbguard/Devices',
 'org.usbguard.Devices',
 'DevicePolicyChanged',
 (45,
  1,
  1,
  'block id 1050:0407 serial "" name "Yubikey 4 OTP+U2F+CCID" hash '
  '"hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=" parent-hash '
  '"jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=" via-port "1-3" '
  'with-interface { 03:01:01 03:00:00 0b:00:00 }',
  4294967294,
  {'hash': 'hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=',
   'id': '1050:0407',
   'name': 'Yubikey 4 OTP+U2F+CCID',
   'parent-hash': 'jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=',
   'serial': '',
   'via-port': '1-3',
   'with-interface': '03:01:01 03:00:00 0b:00:00'}))
{'full_text': 'USBGuard: Yubikey 4 OTP+U2F+CCID', 'urgent': True, 'cached_until': -1}
(':1.7',
 '/org/usbguard/Devices',
 'org.usbguard.Devices',
 'DevicePolicyChanged',
 (45,
  1,
  0,
  'allow id 1050:0407 serial "" name "Yubikey 4 OTP+U2F+CCID" hash '
  '"hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=" parent-hash '
  '"jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=" via-port "1-3" '
  'with-interface { 03:01:01 03:00:00 0b:00:00 }',
  4294967295,
  {'hash': 'hGeSuup3n4/Ky5yJPttU/FeUGqAxIK8xqKKhdVj7ZYw=',
   'id': '1050:0407',
   'name': 'Yubikey 4 OTP+U2F+CCID',
   'parent-hash': 'jEP/6WzviqdJ5VSeTUY8PatCNBKeaREvo2OqdplND/o=',
   'serial': '',
   'via-port': '1-3',
   'with-interface': '03:01:01 03:00:00 0b:00:00'}))
```

Reply:
`via-port` can maybe be proposal as placeholder, if you know your ports...
usb `id` too, if you want to make a usb lab machine ...

Reply:
I'll do placeholders init list, and test store values from current `args[4]` if required. As wwan. Will try managing several next.